code/modules/mob/living/simple_animal/parrot.dm

```diff
@@ -876,11 +891,21 @@
 
 /mob/living/simple_animal/parrot/Poly/proc/Read_Memory()
 	var/savefile/S = new /savefile("data/npc_saves/Poly.sav")
-	S["phrases"] 	>> speech_buffer
+	S["phrases"] 			>> speech_buffer
+	S["rounds survived"]	>> rounds_survived
+	S["longest survival"]	>> longest_survival
+
 	if(isnull(speech_buffer))
 		speech_buffer = list()
 
 /mob/living/simple_animal/parrot/Poly/proc/Write_Memory()
 	var/savefile/S = new /savefile("data/npc_saves/Poly.sav")
-	S["phrases"] 	<< speech_buffer
+	S["phrases"] 			<< speech_buffer
+	if(health > 0)
+		S["rounds survived"]	<< rounds_survived + 1
+		if(rounds_survived == longest_survival)
+			S["longest survival"]	<< longest_survival + 1

```

Comment:
nope, if health is 0 there's no way poly has increased his longest survival streak (cause he's dead)